packages/swapper/src/types.ts

```diff
@@ -101,6 +101,9 @@ type CommonTradeInput = {
   affiliateBps: string
   allowMultiHop: boolean
   slippageTolerancePercentageDecimal?: string
+  // KeepKey is rugged and requires a bandaid to disable affiliate fee for EVM sell assets, see https://github.com/shapeshift/web/issues/4518
+  // Since we have a single shared input across all swappers, we need to pass this as feature detection to monkey patch affiliate bps to 0
+  isKeepKey: boolean
```

Comment:
Rather than letting swappers know about keepkey, it might be better to manage the bps mutation in the api when we generate the input params. This way we're more dry and wont miss swappers.

Looks like thorchain swapper is missed as an example (long tail means we can get a thorchain swapper quote FOX -> USDT with affiliate 29bps fee):

![image](https://github.com/shapeshift/web/assets/125113430/ff0f8a6b-255d-4802-92e6-c9eed3399259)
![image](https://github.com/shapeshift/web/assets/125113430/8b25d1f4-e03f-408c-97db-a97904ba93b3)

Reply:
As discussed, we *do* want to exclude THOR from the monkey patch as that's the actual fix here, and we do want to exclude other swappers in a follow-up as well, a.k.a remove the bandaid entirely once we have capacity for properly testing this

Reply:
as discussed, this will be possible once we have wallet selectors available in redux after changes for https://github.com/shapeshift/web/issues/5946 go in. That will allow us to check wallet type without trying to pass wallet into apis